File: pios-app-ui/src/views/component-search/Component.ts
import Manufacturer from '~/views/component-search/Manufacturer';
import Type from '~/views/component-search/Type';

interface Component {
  id: number;
  name: string;
  price: number;
  type: Type;
  data: string;
  imageBase64: string | undefined;
  manufacturer: Manufacturer;
  rating?: number;
  reviewCount: number;
  reviewed?: boolean;
}

export default Component;

File: pios-app-service/src/main/java/hr/tvz/pios/modul/component/Component.java
package hr.tvz.pios.modul.component;

import hr.tvz.pios.common.Type;
import hr.tvz.pios.modul.manufacturer.Manufacturer;
import java.math.BigDecimal;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Component {

  Long id;

  String name;

  BigDecimal price;

  Type type;

  String data;

  String imageBase64;

  Manufacturer manufacturer;

  Double rating;

  Integer reviewCount;

  Boolean reviewed;
}
